x/merkledb/trieview.go

```diff
@@ -279,22 +276,22 @@ func (t *trieView) calculateNodeIDsHelper(ctx context.Context, n *node, eg *errg
 		}
 
 		wg.Add(1)
-		updateChild := func() {
+		calculateChildID := func() {
 			defer wg.Done()
 
-			t.calculateNodeIDsHelper(ctx, childNodeChange.after, eg)
+			t.calculateNodeIDsHelper(ctx, childNodeChange.after)
 
 			// Note that this will never block
 			updatedChildren <- childNodeChange.after
 		}
 
 		// Try updating the child and its descendants in a goroutine.
-		if ok := eg.TryGo(func() error {
-			updateChild()
-			return nil
-		}); !ok {
+		if ok := t.db.calculateNodeIDsSema.TryAcquire(1); ok {
+			go calculateChildID()
+			t.db.calculateNodeIDsSema.Release(1)
```

Comment:
shouldn't this be part of the goroutine and deferred?

Reply:
fixed